src/main/java/org/jabref/gui/fieldeditors/contextmenu/EditorMenus.java

```diff
@@ -14,42 +15,73 @@
 import javafx.scene.control.Tooltip;
 
 import org.jabref.gui.actions.CopyDoiUrlAction;
+import org.jabref.gui.fieldeditors.EditorTextArea;
 import org.jabref.logic.formatter.bibtexfields.NormalizeNamesFormatter;
 import org.jabref.logic.l10n.Localization;
 
+/**
+ * Provides context menus for the text fields of the entry editor. Note that we use {@link Supplier} to prevent an early
+ * instantiation of the menus. Therefore, they are attached to each text field but instantiation happens on the first
+ * right-click of the user in that field. The late instantiation is done by {@link
+ * EditorTextArea#addToContextMenu(java.util.function.Supplier)}.
+ */
 public class EditorMenus {
 
-    public static List<MenuItem> getDefaultMenu(TextArea textArea) {
-        List<MenuItem> menuItems = new ArrayList<>(5);
-        menuItems.add(new CaseChangeMenu(textArea.textProperty()));
-        menuItems.add(new ConversionMenu(textArea.textProperty()));
-        menuItems.add(new SeparatorMenuItem());
-        menuItems.add(new ProtectedTermsMenu(textArea));
-        menuItems.add(new SeparatorMenuItem());
-        return menuItems;
+    /**
+     * The default menu that contains functions for changing the case of text and doing several conversions.
+     *
+     * @param textArea text-area that this menu will be connected to
+     * @return default context menu available for most text fields
+     */
+    public static Supplier<List<MenuItem>> getDefaultMenu(TextArea textArea) {
+        return () -> {
```

Comment:
I'm not sure what is more readable: Hiding the supplier interface here like you did or instead writing something like `addToContextMenu(() -> getDefaultMenu(textArea))`. @JabRef/developers any opinions on this?

Reply:
I think I am in favor of the second solution, i.e. putting the supplier into `EditorTextArea.addToContextMenu()`. Reasoning: With the current solution, the lazy initialization is dispersed over two classes. Changes are required in `EditorTextArea` and `EditorMenus`. If I understand everything correctly, we could build the supplier inside `addToContextMenu()` only. That would be a very localized change that hides the lazy initialization as far as possible.

My perception might be wrong, though. Please correct me if this is the case.